### PR TITLE
Don't lock account until a valid ppolicy with pwdLockout=TRUE is associated (#45)

### PR DESCRIPTION
**Pull request by davidcoutadeur**
Don't lock account until a valid ppolicy with pwdLockout=TRUE is associated (#45)

### CODE REVIEW

File: htdocs/lockaccount.php [hunk 1]
```diff
@@ -24,7 +24,46 @@
...
+
+        # Search active password policy
+        $pwdPolicy = "";
+        if (isset($user_entry[0]['pwdpolicysubentry'][0])) {
+            $pwdPolicy = $user_entry[0]['pwdpolicysubentry'][0];
+        } elseif (isset($ldap_default_ppolicy)) {
+            $pwdPolicy = $ldap_default_ppolicy;
+        }
+
+        $ppolicy_entry = "";
```
Comment (coudot) [inline htdocs/lockaccount.php:46]: I see no need to initialize it here, as it is not used outside the above code.

File: htdocs/lockaccount.php [hunk 2]
```diff
@@ -24,7 +24,46 @@
     $ldap = $ldap_connection[0];
     $result = $ldap_connection[1];
 
-    if ($ldap) {
+
+    # Consider pwdLockout = false by default
+    $pwdLockout = "false";
```
Comment (coudot) [inline htdocs/lockaccount.php:29]: Should be better to use a real PHP boolean here (true/false without quotes)

File: htdocs/lockaccount.php [hunk 3]
```diff
@@ -24,7 +24,46 @@
...
+
+        # Search pwdLockout attribute
+        if ($pwdPolicy) {
+                $search_ppolicy = ldap_read($ldap, $pwdPolicy, "(objectClass=pwdPolicy)", array('pwdlockout'));
+
+                if ( $errno ) {
+                    error_log("LDAP - PPolicy search error $errno  (".ldap_error($ldap).")");
+                } else {
+                    $ppolicy_entry = ldap_get_entries($ldap, $search_ppolicy);
+                    $pwdLockout = strtolower($ppolicy_entry[0]['pwdlockout'][0]);
```
Comment (coudot) [inline htdocs/lockaccount.php:56]: Transform in a PHP boolean

File: htdocs/lockaccount.php [hunk 4]
```diff
@@ -24,7 +24,46 @@
...
+                    if(! isset($pwdLockout) or $pwdLockout == "" or $pwdLockout == "false" )
```
Comment (coudot) [inline htdocs/lockaccount.php:57]: You should just check if pwdLockout is false after it has been converted to a PHP boolean

File: htdocs/lockaccount.php [hunk 5]
```diff
@@ -24,7 +24,46 @@
...
+                    {
+                        error_log("No pwdLockout or pwdLockout=FALSE in associated ppolicy: ".$pwdPolicy.". Account locking disabled");
+                    }
+                }
+        }
+    }
+
+    # apply the modification only if a password policy set with pwdLockout=TRUE is associated to the account
+    if ($ldap and isset($pwdLockout) and $pwdLockout == "true") {
```
Comment (coudot) [inline htdocs/lockaccount.php:66]: Use PHP boolean

**Review by coudot — CHANGES_REQUESTED** [review]

**Comment by davidcoutadeur** [comment]
Still working on the frontend part to forbid lockout

**Comment by davidcoutadeur** [comment]
feature seems ok for me. You can review further commits and merge if ok for you

**Review by coudot — APPROVED** [review]

**Pull request merged**